libs/src/sdk/oauth/OAuth.ts

```diff
@@ -140,25 +148,61 @@ export class OAuth {
     )
   }
 
-  login() {
-    if (!this.config.appName) {
+  async isWriteAccessGranted(params: IsWriteAccessGrantedRequest) {
+    const { userId, apiKey } = parseRequestParameters(
+      'isWriteAccessGranted',
+      IsWriteAccessGrantedSchema
+    )(params)
+    if (!this.apiKey && !apiKey) {
+      this._surfaceError(
+        'Need to init Audius SDK with API key or pass in API Key directly to oauth.isWriteAccessGranted.'
+      )
+    }
+    const authorizedApps = await this.config.usersApi.getAuthorizedApps({
+      id: userId
+    })
+
+    const foundIndex = authorizedApps.data?.findIndex(
+      (a) =>
+        a.address.toLowerCase() ===
+        `0x${(apiKey || this.apiKey)!.toLowerCase()}`
+    )
+    return foundIndex !== undefined && foundIndex > -1
+  }
+
+  login(scope: OAuthScope = 'read') {
+    if (!this.config.appName && !this.apiKey) {
       this._surfaceError('App name not set (set with `init` method).')
       return
     }
+    if (scope === 'write' && !this.apiKey) {
+      this._surfaceError(
+        'Need to init Audius SDK with API key in order to use oauth.login with scope = write'
+      )
+    }
     if (!this.loginSuccessCallback) {
       this._surfaceError(
         'Login success callback not set (set with `init` method).'
       )
       return
     }
+    if (scope !== 'read' && scope !== 'write') {
```

Comment:
write just means both (since technically anyone can read anyone's account), is that confusing? wonder if there's a better way to specify the difference...

Reply:
I think that's fine, but I think often apis let you specify multiple scopes so that works too